### ISSUE TITLE
Phantom display on Plasma 6.1 Wayland on Nvidia

### ISSUE THREAD

**Issue by aidalgol** [NixOS/nixpkgs#321603]
### Describe the bug
~~Plasma 6.1 on an Nvidia RTX 30 series desktop GPU displays a mostly-black screen on login, with one window partially visible.~~

See https://github.com/NixOS/nixpkgs/issues/321603#issuecomment-2187846953.  Display Configuration shows an extra display named "Unknown-1" that does not correspond to any physical display, and is thus invisible.

### Steps To Reproduce
1. Put the following in the NixOS system configuration:
```nix
  boot.kernelModules = [
    "nvidia"
    "nvidia_uvm"
    "nvidia_modeset"
    "nvidia_drm"
  ];

  hardware.opengl = {
    enable = true;
    driSupport32Bit = true;
  };
  hardware.nvidia = {
    modesetting.enable = true;
    nvidiaPersistenced = true;
    package = config.boot.kernelPackages.nvidiaPackages.beta;
    open = true;
  };
  # Force Vulkan to only try using Nvidia proprietary driver.
  environment.sessionVariables.VK_DRIVER_FILES = "/run/opengl-driver/share/vulkan/icd.d/nvidia_icd.x86_64.json";

  services = {
    xserver = {
      enable = true;
      videoDrivers = ["nvidia"];
      wacom.enable = true;
    };
    displayManager.sddm.enable = true;
    desktopManager.plasma6.enable = true;
  };
  # Enable NixOS Electron/Chromium Wayland support.
  environment.sessionVariables.NIXOS_OZONE_WL = "1";
```
2. Rebuild and reboot.
3. Log in from display manager.

### Expected behavior
A Plasma session starts normally.

### Screenshots
![displays](https://github.com/NixOS/nixpkgs/assets/2313201/d47a8d52-cf7b-437f-9a6e-342036bf59b1)

### Additional context
My system config is flake-based, on commit `c00d587b1a1afbf200b1d8f0b0e4ba9deb1c7f0e` of `nixpkgs`.

AppArmor is enabled.

Output from `journalctl -b`, run from a virtual console after logging in to a Plasma session: [journal-boot-plasma61-wayland.log](https://github.com/user-attachments/files/15933921/journal-boot-plasma61-wayland.log)

### Notify maintainers
@NixOS/qt-kde 

### Metadata
```console
❯ nix shell nixpkgs#nix-info -c nix-info -m
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.34, NixOS, 24.11 (Vicuna), 24.11.20240618.c00d587`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.90.0-rc1-pre20240615-253546d`
 - nixpkgs: `/home/aidan/.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by alexanderzjs** [comment]
Not sure, but maybe you need drm modeset? https://wiki.archlinux.org/title/NVIDIA#DRM_kernel_mode_setting

**Comment by aidalgol** [comment]
That's set via `hardware.nvidia.modesetting.enable = true;` and I just confirmed that `cat /sys/module/nvidia_drm/parameters/modeset` prints `Y`.

**Comment by VeilSilence** [comment]
Try set `open =false` It's working for me atleast.

**Comment by aidalgol** [comment]
Setting `hardware.nvidia.open = false;` made no difference, but purely out of frustration, I shook the mouse and happened to hit a corner set to show the overview.  I then tried bringing up the launcher to open the display settings, and I saw this.
![Screenshot_20240625_095705](https://github.com/NixOS/nixpkgs/assets/2313201/1730fc2b-16fc-4cb4-ad8c-58320df14b20)
That display named "Unknown-1" is not a real display, so I cannot see anything on it, and that was where all my application windows were hiding.  So Plasma 6.1 Wayland on Nvidia works, but there is a bug.
